File: 4week/Soom2/Programmers_42885.py
def solution(people, limit):
    people.sort()
    left = 0
    right = len(people) - 1
    answer = 0

    while left <= right:
        if left == right:
            answer += 1
            break
        if people[left] + people[right] <= limit:
            answer += 1
            right -= 1
            left += 1
        else:
            answer += 1
            right -= 1

    return answer


print(solution([70, 50, 80, 50], 100))

File: 5week/PHS00/Programmers_42885.java
package PHS00;

import java.util.*;
public class Programmers_42885 {

    class Solution {
        public int solution(int[] people, int limit) {
            int answer = 0;
            Arrays.sort(people);
            int lp = 0;
            int rp = people.length - 1;
            while(lp <= rp){
                if(people[lp] + people[rp] <= limit){
                    answer++;
                    lp++;
                    rp--;
                }else{
                    rp--;
                    answer++;
                }

            }

            return answer;
        }
    }
}
